diff --git a/libtiff/CMakeLists.txt b/libtiff/CMakeLists.txt
index a8aa0c32..5453c8b1 100755
--- a/libtiff/CMakeLists.txt
+++ b/libtiff/CMakeLists.txt
@@ -184,9 +184,9 @@ if(WEBP_SUPPORT)
   target_link_libraries(tiff PRIVATE WebP::webp)
   string(APPEND tiff_requires_private " libwebp")
 endif()
-if(CMath_LIBRARY)
-  target_link_libraries(tiff PRIVATE CMath::CMath)
-  list(APPEND tiff_libs_private_list "${CMath_LIBRARY}")
+find_library(MATH_LIBRARY m)
+if(MATH_LIBRARY)
+    target_link_libraries(tiff PUBLIC ${MATH_LIBRARY})
 endif()
 
 set(tiff_libs_private_list "${tiff_libs_private_list}" PARENT_SCOPE)
