diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 6c84b7b..07f7677 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -129,7 +129,8 @@ endif()
 
 # Detect and remove GCC _FORTIFY_SOURCE
 if(CMAKE_C_COMPILER_ID STREQUAL "GNU")
-  execute_process(COMMAND ${CMAKE_C_COMPILER} ${CMAKE_C_FLAGS} -E -Og -dM -x c ${dev_null} OUTPUT_VARIABLE gcc_macros)
+  separate_arguments(C_FLAGS_LIST UNIX_COMMAND "${CMAKE_C_FLAGS}")
+  execute_process(COMMAND ${CMAKE_C_COMPILER} ${C_FLAGS_LIST} -E -Og -dM -x c ${dev_null} OUTPUT_VARIABLE gcc_macros)
   string(FIND ${gcc_macros} "_FORTIFY_SOURCE" fortify)
   if(fortify GREATER -1)
     message(WARNING "GCC _FORTIFY_SOURCE is known to cause problem with Scotch and has been disabled")
